ui/src/components/Banner.jsx

```diff
@@ -0,0 +1,272 @@
+// The ring size 17*17 = 289 is close to a multiple of 41 (41*7 = 287)
+
+  vec3 x = 2.0 * fract(p * C.www) - 1.0;
+  vec3 h = abs(x) - 0.5;
+  vec3 ox = floor(x + 0.5);
+  vec3 a0 = x - ox;
+
+// Normalise gradients implicitly by scaling m
+// Approximation of: m *= inversesqrt( a0*a0 + h*h );
+  m *= 1.79284291400159 - 0.85373472095314 * ( a0*a0 + h*h );
+
+// Compute final noise value at P
+  vec3 g;
+  g.x  = a0.x  * x0.x  + h.x  * x0.y;
+  g.yz = a0.yz * x12.xz + h.yz * x12.yw;
+  return 130.0 * dot(m, g);
+}
+
+float staticV(vec2 uv) {
+    float staticHeight = snoise(vec2(9.0,iTime*1.2+3.0))*0.3+5.0;
+    float staticAmount = snoise(vec2(1.0,iTime*1.2-6.0))*0.1+0.3;
+    float staticStrength = snoise(vec2(-9.75,iTime*0.6-3.0))*2.0+2.0;
+	return (1.0-step(snoise(vec2(5.0*pow(iTime,2.0)+pow(uv.x*7.0,1.2),pow((mod(iTime,100.0)+100.0)*uv.y*0.3+3.0,staticHeight))),staticAmount))*staticStrength;
+}
+
+
+void mainImage( out vec4 fragColor, in vec2 fragCoord )
+{
+	vec2 uv =  fragCoord.xy/u_imageResolution.xy;
+
+	float jerkOffset = (1.0-step(snoise(vec2(iTime*1.3,5.0)),0.8))*0.05;
+
+	float fuzzOffset = snoise(vec2(iTime*15.0,uv.y*80.0))*0.003;
+	float largeFuzzOffset = snoise(vec2(iTime*1.0,uv.y*25.0))*0.004;
+
+    float vertMovementOn = (1.0-step(snoise(vec2(iTime*0.2,8.0)),0.4))*vertMovementOpt;
+    float vertJerk = (1.0-step(snoise(vec2(iTime*1.5,5.0)),0.6))*vertJerkOpt;
+    float vertJerk2 = (1.0-step(snoise(vec2(iTime*5.5,5.0)),0.2))*vertJerkOpt;
+    float yOffset = abs(sin(iTime)*4.0)*vertMovementOn+vertJerk*vertJerk2*0.3;
+    float y = mod(uv.y+yOffset,1.0);
+
+
+	float xOffset = (fuzzOffset + largeFuzzOffset) * horzFuzzOpt;
+
+    float staticVal = 0.0;
+
+    for (float y = -1.0; y <= 1.0; y += 1.0) {
+        float maxDist = 5.0/200.0;
+        float dist = y/200.0;
+    	staticVal += staticV(vec2(uv.x,uv.y+dist))*(maxDist-abs(dist))*1.5;
+    }
+
+    staticVal *= bottomStaticOpt;
+
+	float red 	=   texture2D(	u_image, 	vec2(uv.x + xOffset -0.01*rgbOffsetOpt,y)).r+staticVal;
+	float green = 	texture2D(	u_image, 	vec2(uv.x + xOffset,	  y)).g+staticVal;
+	float blue 	=	texture2D(	u_image, 	vec2(uv.x + xOffset +0.01*rgbOffsetOpt,y)).b+staticVal;
+
+	vec3 color = vec3(red,green,blue);
+	float scanline = sin(uv.y*800.0)*0.04*scalinesOpt;
+	color -= scanline;
+
+	fragColor = vec4(color,1.0);
+}
+
+void main() {
+  mainImage(gl_FragColor, gl_FragCoord.xy*vec2(1, -1));
+}
+`;
+
+function createProgram(gl, vertexShader, fragmentShader) {
+  var program = gl.createProgram();
+  gl.attachShader(program, vertexShader);
+  gl.attachShader(program, fragmentShader);
+  gl.linkProgram(program);
+  var success = gl.getProgramParameter(program, gl.LINK_STATUS);
+  if (success) {
+    return program;
+  }
+
+  console.log(gl.getProgramInfoLog(program));
+  gl.deleteProgram(program);
+}
+
+function setRectangle(gl, x, y, width, height) {
+  var x1 = x;
+  var x2 = x + width;
+  var y1 = y;
+  var y2 = y + height;
```

Comment:
this is actually code I wrote (or copy pasted actually), but we could change those to `const`